winter/core/annotation_decorator.py

```diff
@@ -0,0 +1,47 @@
+import inspect
+import types
+import typing
+
+from .component import Component
+from .component_method import ComponentMethod
+
+
+def annotate(annotation: typing.Any) -> typing.Union[typing.Type, ComponentMethod]:
+
+    def wrapper(func_or_cls: typing.Union[typing.Type, types.FunctionType, ComponentMethod]):
+
+        if isinstance(func_or_cls, ComponentMethod) or isinstance(func_or_cls, types.FunctionType):
+            decorator = annotate_method(annotation)
+        elif inspect.isclass(func_or_cls):
+            decorator = annotate_class(annotation)
+        else:
+            raise ValueError(f'Need function or class. Got: {func_or_cls}')
+        return decorator(func_or_cls)
+    return wrapper
+
+
+def annotate_class(value: typing.Any) -> typing.Type:
+
+    def wrapper(cls: typing.Type):
+        if inspect.isclass(cls):
+            component = Component.get_by_cls(cls)
```

Comment:
Not an-approve stopper: maybe pass cls to `Component.get_by_cls` as it is ant let him to validate the parameter?